insights/parsers/grubenv.py

```diff
@@ -90,3 +94,61 @@ def has_tuned_params(self):
     @property
     def tuned_params(self):
         return self.get("tuned_params", "")
+
+
+@parser(Specs.grubenv)
+class Grub2EditenvList(GrubEnv):
+    """
+    Parses the ``grub2-editenv list`` command and returns a dict
+    of the grubenv variables. The command output without error
+    message is same as the content in ``/boot/grub2/grubenv``.
+
+    Sample output of the command::
+
+        saved_entry=295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64
+        kernelopts=root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200
+        boot_success=0
+        boot_indeterminate=2
+        tuned_params=transparent_hugepages=never
+        tuned_initrd=
+
+    Examples:
+
+        >>> type(grub2_editenv_list)
+        <class 'insights.parsers.grubenv.Grub2EditenvList'>
+        >>> grub2_editenv_list.has_kernelopts
+        True
+        >>> grub2_editenv_list.kernelopts
+        'root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200'
+        >>> grub2_editenv_list.has_tuned_params
+        True
+        >>> grub2_editenv_list.tuned_params
+        'transparent_hugepages=never'
+        >>> grub2_editenv_list['saved_entry']
+        '295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64'
+    """
+
+    def __init__(self, context):
+        super(Grub2EditenvList, self).__init__(context)
```

Comment:
It seems "errors" is the only difference between the two parsers, it would be better to add a new base class and make them two inherit from the base.

Reply:
@xiangce Updated